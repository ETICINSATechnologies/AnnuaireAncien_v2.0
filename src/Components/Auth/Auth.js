class Auth {
    static connect(jwt){
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(){
        sessionStorage.removeItem('token')
    }

    static isConnected(){
        return !!sessionStorage.getItem('token');
    }

    static getToken() {
        return sessionStorage.getItem('token')
    }

    static getPayload(){
        return sessionStorage.getItem('token');
    }

    static addCorrectButton(buttons){
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
