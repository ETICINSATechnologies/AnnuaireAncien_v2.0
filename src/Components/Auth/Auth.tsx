class Auth {
    static connect(jwt: string) {
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(): void {
        sessionStorage.removeItem('token')
    }

    static isConnected(): boolean {
        return !!sessionStorage.getItem('token');
    }

    static getToken(): string {
        return sessionStorage.getItem('token') as string;
    }

    static addCorrectButton(buttons: Array<string>) {
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
