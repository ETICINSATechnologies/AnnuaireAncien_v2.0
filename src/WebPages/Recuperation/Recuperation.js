import React, {Component} from 'react';
import './Recuperation.css';
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import Auth from "../../Components/Auth/Auth";

class Recuperation extends Component {
    render() {
        let activeButton = Auth.addCorrectButton(["home"]);

        return (
            <React.Fragment>
                <Header/>
                <Nav buttons={activeButton}> </Nav>
                <div className="Recuperation">
                    <section className="Recuperation_section">
                        <div className="div1">
                            Afin de récupérer vos identifiants, veuillez contacter par mail l'administrateur de ce site
                            en
                            utilisant
                            l'adresse mail suivante :
                        </div>
                        <button className="div2"> contact@etic-insa.com</button>
                        <div className="div3"> adresse copié !</div>
                    </section>
                </div>
            </React.Fragment>
        );
    }
}

export default Recuperation;
