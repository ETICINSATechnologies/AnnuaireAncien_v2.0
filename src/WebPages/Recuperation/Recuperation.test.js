import React from 'react';

import Recuperation from './Recuperation';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Auth from '../../Components/Auth/Auth';

Enzyme.configure({adapter: new Adapter()});

describe( 'Recuperation component', () => {
    it('renders without crashing', () => {
        const tree = renderer.create(<Recuperation />).toJSON();
        expect(tree).toMatchSnapshot();
        }
    );

    // it('renders a message', () => {
    //     expect(Enzyme.shallow(<Recuperation />).find('#Recuperation').length).toEqual(1)
    // });

    it('Recuperation unit test', () => {
        expect(
            Enzyme.shallow(<Recuperation/>)
                .find('div.Recuperation')
                .length
        ).toBe(1)
        // const wrapper = Enzyme.shallow(<Recuperation/>)
        //     .find('div.Recuperation')
        //     .length
        //     )
        // console.log(wrapper.debug());
    });

    it('Recuperation user connected button test', () => {
        Auth.isConnected = jest.fn(() => true);
        const wrapper = Enzyme.shallow(<Recuperation />);
        // expect(wrapper.find({ href: '/'})).toBe(1);
        const expected = ["home","disconnection"];
        expect( Auth.addCorrectButton(["home"]) ).toEqual(expected);
        }
    );

    it('Recuperation user not connected button test', () => {
            Auth.isConnected = jest.fn(() => false);
            const wrapper = Enzyme.shallow(<Recuperation />);
            const expected = ["home","connection"];
            expect( Auth.addCorrectButton(["home"]) ).toEqual(expected);
        }
    );

});