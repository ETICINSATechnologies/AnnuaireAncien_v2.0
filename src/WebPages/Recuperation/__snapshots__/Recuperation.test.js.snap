// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recuperation component renders without crashing 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/"
      >
        accueil
      </a>
      <a
        className="button"
        href="/connection"
      >
        se connecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <div
    className="Recuperation"
  >
    <section
      className="Recuperation_section"
    >
      <div
        className="div1"
      >
        Afin de récupérer vos identifiants, veuillez contacter par mail l'administrateur de ce site en utilisant l'adresse mail suivante :
      </div>
      <button
        className="div2"
      >
         contact@etic-insa.com
      </button>
      <div
        className="div3"
      >
         adresse copié !
      </div>
    </section>
  </div>,
]
`;
